_ml-commons-plugin/agents-tools/tools/rag-tool.md

```diff
@@ -0,0 +1,132 @@
+---
+layout: default
+title: RAG tool
+has_children: false
+has_toc: false
+nav_order: 65
+parent: Tools
+grand_parent: Agents and tools
+---
+
+<!-- vale off -->
+# RAG tool
+**Introduced 2.12**
+{: .label .label-purple }
+<!-- vale on -->
+
+This is an experimental feature and is not recommended for use in a production environment. For updates on the progress of the feature or if you want to leave feedback, see the associated [GitHub issue](https://github.com/opensearch-project/ml-commons/issues/1161).    
+{: .warning}
+
+The `RAGTool` performs retrieval-augmented generation (RAG). For more information about RAG, see [conversational search]({{site.url}}{{site.baseurl}}/search-plugins/conversational-search/).
+
+Retrieval-augmented generation calls a large language model (LLM) and supplements the LLM knowledge by providing relevant OpenSearch documents along with the user question. To retrieve relevant documents from an OpenSearch index, you'll need a text embedding model that facilitates vector search. 
+
+## Before you start
+
+To register and deploy a text embedding model and an LLM and ingest data into an index, perform Steps 1--5 of the [Agents and tools tutorial]({{site.url}}{{site.baseurl}}/ml-commons-plugin/agents-tools/agents-tools-tutorial/).
+
```

Comment:
should link register and deploy a sparse embedding model from the neural sparse tool page, and prepare sparse embedding instructions https://opensearch.org/docs/latest/search-plugins/neural-sparse-search/